change/@ni-nimble-blazor-eed57bcb-a0ee-4959-bdc1-c9e233274889.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
@munteannatan this PR was moved into owner's review but does not have a passing build so is incomplete, what is the intention here?

Reply:
Sorry, my mistake.

The build is failing because the test with JavaScript evaluation is behaving inconsistently in the command line, vs editor and GitHub CI. I was hoping you would have an idea regarding a fix that I mentioned in my other comment.

I put it out of the draft because it seemed that was the last signaled issue and I was thinking you will get notified, but I should have just mentioned you.

If you think there are more issues to consider I will put it back in draft.

Reply:
I see. I unfortunately don't have too many specific thoughts on what the issue may be.

It could be timing, maybe try different waits before reading the canvas.
It could be visibility, maybe the canvas behaves different if it is not visible in the browser windows viewport.
It could be chrome setup, in our karma tests we pass [lots of flags](https://github.com/ni/nimble/blob/b61e3164866c8703fad31b2673fc88ebb694c6fd/packages/nimble-components/karma.conf.js#L17) to control different settings for avoiding background timing, animations, etc. Maybe something impacts how canvas behaves?

Reply:
I will try to find the issue then, thanks for the hint!

Reply:
Another thought is that for timing related issues, particularly in the table which also has batch updates, we added a private / internal method called [`processPendingUpdates`](https://github.com/ni/nimble/blob/3d77b048ac2f4cf191eb8464f1c014882e1d7d73/packages/nimble-components/src/table/index.ts#L706) that can way for any currently batched operations to be resolved. This pattern may be particularly useful when threaded /worker support is added. I would probably not recommend that it be made a public api but could be useful as private / internal for tests.

Reply:
the problem was that the eval was called before the canvas was there. I added an assert to wait for the canvas to be visible